client/state/plugins/wporg/selectors.js

```diff
@@ -24,4 +24,36 @@ const isFetched = function( state, pluginSlug ) {
 	return !! plugin.fetched;
 };
 
-export default { getPlugin, isFetching, isFetched };
+const canFetchList = function( state, category, page, searchTerm ) {
+	const { lists } = state.plugins.wporg;
+	if ( searchTerm && lists.fetching.search ) {
+		return false;
+	}
+	if ( lists.fetching[ category ] ) {
+		return false;
+	}
+
+	if ( lists.lastFetchedPage[ category ] >= page ) {
+		return false;
+	}
+
+	return true;
+};
+
+const isFetchingList = function( state, category ) {
+	return !! state.plugins.wporg.lists.fetching[ category ];
+};
+
+const getCurrentSearchTerm = function( state ) {
+	return state.plugins.wporg.lists.currentSearchTerm;
+};
+
+const getList = function( state, category ) {
+	return state.plugins.wporg.lists.fullLists[ category ] ? state.plugins.wporg.lists.fullLists[ category ] : [];
```

Comment:
I'm fine with leaving it as-is, as long as there's no way `state.plugins.wporg.lists.fullLists[ category ]` would throw an error because a prop in the chain is not defined.